backend/src/ee/services/pki-acme/pki-acme-service.ts

```diff
@@ -732,6 +739,10 @@ export const pkiAcmeServiceFactory = ({
           throw new AcmeBadCSRError({ message: "Invalid CSR: Common name + SANs mismatch with order identifiers" });
         }
 
+        if (!profile.caId) {
+          throw new NotFoundError({ message: "Self-signed certificates are not supported for ACME enrollment" });
+        }
```

Comment:
isn't this check redundant with the one above? and shouldn't it be BadRequestError?